ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/ListReceipts.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright contributors to Hyperledger Besu
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.core;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.jetbrains.annotations.NotNull;
+
+public class ListReceipts extends ArrayList<TransactionReceipt> {
```

Comment:
Please use a "tuple" object (one field is RLP, one field is the List<TransactionReceipts>) instead of extending a standard collections object.  This ties us deeply to one collection implementation.

Reply:
I added new fields to this objects. using the list avoid to change a lot of code but I will check if I can update all the part and creating a new object instead of extending the list